utilities/deep_inspection.py

```
@@ -96,6 +107,107 @@ def create_di_connection_secret(
+
+    Raises:
+        TimeoutExpiredError: If snapshot does not appear within timeout.
+    """
+    vm = source_provider.get_vm_by_name(query=vm_name)
+    for snapshots in TimeoutSampler(
+        wait_timeout=timeout,
+        sleep=5,
+        func=lambda: [s for s in source_provider.list_snapshots(vm) if s.name == DI_SNAPSHOT_NAME],
+    ):
+        if snapshots:
+            LOGGER.info(f"DI snapshot '{DI_SNAPSHOT_NAME}' found on VM '{vm_name}'")
```

Ruff (0.15.20)

[warning] Logging statement uses f-string

(G004)